pyproject.toml

```
@@ -29,13 +29,13 @@ dynamic = ["description", "version"]
 requires-python = ">=3.7"
 dependencies = [
     "click>=7.1,<9",
-    "docutils>=0.15,<0.19", # report issue for >=0.18,<0.20 until https://github.com/mcmtroffaes/sphinxcontrib-bibtex/issues/322 is fixed
+    "docutils>=0.15,!=0.18.*,!=0.19.*",
     "Jinja2",
     "jsonschema<5",
     "linkify-it-py~=2.0.0",
     "myst-nb~=0.17.1",
     "pyyaml",
-    "sphinx>=4,<6",
+    "sphinx>=4,!=6.*,!=7.0.0",
```

or perhaps

```suggestion
    "sphinx>=4,!=6.*,!=7.0.0,<8",
```

Note that `sphinx==7.0.1` is the first version which permits `docutils>0.19`.
However I don't know if it breaks `jupyter-book`. This needs testing :)